igzip/huff_codes.c

```diff
@@ -728,39 +728,21 @@ void isal_update_histogram_base(uint8_t * start_stream, int length,
 static uint32_t convert_dist_to_dist_sym(uint32_t dist)
 {
 	assert(dist <= 32768 && dist > 0);
-	if (dist <= 2)
-		return dist - 1;
-	else if (dist <= 4)
-		return 0 + (dist - 1) / 1;
-	else if (dist <= 8)
-		return 2 + (dist - 1) / 2;
-	else if (dist <= 16)
-		return 4 + (dist - 1) / 4;
-	else if (dist <= 32)
-		return 6 + (dist - 1) / 8;
-	else if (dist <= 64)
-		return 8 + (dist - 1) / 16;
-	else if (dist <= 128)
-		return 10 + (dist - 1) / 32;
-	else if (dist <= 256)
-		return 12 + (dist - 1) / 64;
-	else if (dist <= 512)
-		return 14 + (dist - 1) / 128;
-	else if (dist <= 1024)
-		return 16 + (dist - 1) / 256;
-	else if (dist <= 2048)
-		return 18 + (dist - 1) / 512;
-	else if (dist <= 4096)
-		return 20 + (dist - 1) / 1024;
-	else if (dist <= 8192)
-		return 22 + (dist - 1) / 2048;
-	else if (dist <= 16384)
-		return 24 + (dist - 1) / 4096;
-	else if (dist <= 32768)
-		return 26 + (dist - 1) / 8192;
-	else
-		return ~0;	/* ~0 is an invalid distance code */
-
+	if (dist <= 32768) {
+		uint32_t val;
+#ifdef _MSC_VER
+		unsigned long where;
+		if ((dist > 4) && _BitScanReverse(&where, (dist - 1) / 2))
```

Comment:
I think it would be better to optimize the existing bsr() function to find the msb and use this instead.  Can you look at the PR I pushed for this (#81) and test?  This becomes simpler also.  Something like:
```
uint32_t msb = dist > 4 ? bsr(dist - 1) - 2 : 0;
uint32_t code = (msb * 2) + ((dist - 1) >> msb);
```

Reply:
Nice. I didn't realize that there is a bsr function in huffman.h.
Will follow this way.